### PR TITLE
[SPARK-23828][ML][PYTHON]PySpark StringIndexerModel should have constructor from labels

### PR DESCRIPTION
**Pull request by huaxingao**
## What changes were proposed in this pull request?

The Scala StringIndexerModel has an alternate constructor that will create the model from an array of label strings.  Add the corresponding Python API:

model = StringIndexerModel.from_labels(["a", "b", "c"])

## How was this patch tested?

Add doctest and unit test.

### CODE REVIEW

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -2342,8 +2342,38 @@ def mean(self):
...
+    def getStringOrderType(self):
+        """
+        Gets the value of :py:attr:`stringOrderType` or its default value 'frequencyDesc'.
+        """
+        return self.getOrDefault(self.stringOrderType)
+
+
 @inherit_doc
-class StringIndexer(JavaEstimator, HasInputCol, HasOutputCol, HasHandleInvalid, JavaMLReadable,
+class StringIndexer(JavaEstimator, _StringIndexerParams, HasHandleInvalid, JavaMLReadable,
```
Comment (BryanCutler) [inline python/pyspark/ml/feature.py]: you should move `HasHandleInvalid` to be a trait for `_StringIndexerParam`

**Review by BryanCutler — CHANGES_REQUESTED** [review]
Thanks @huaxingao , could you also add a test to check for default param values?  similar to the check for CountVectorizerModel here https://github.com/apache/spark/blob/master/python/pyspark/ml/tests.py#L2101

Also, it might be good to add unit tests for things like if the list of labels is empty etct, like here https://github.com/apache/spark/blob/master/python/pyspark/ml/tests.py#L743

**Review by BryanCutler — APPROVED** [review]
LGTM, thanks @huaxingao !

**Comment by BryanCutler** [comment]
merged to master

**Comment by huaxingao** [comment]
@BryanCutler Thank you very much for your help!

**Pull request closed without merging**